src/api/v2/compatibility/CompatibleWorkspaceResourceBranchDataProvider.ts

```diff
@@ -0,0 +1,80 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzExtParentTreeItem, AzExtTreeDataProvider, AzExtTreeItem, IFindTreeItemContext, ITreeItemPickerContext } from '@microsoft/vscode-azext-utils';
+import type { WorkspaceResource } from '@microsoft/vscode-azext-utils/hostapi';
+import * as vscode from 'vscode';
+import type { BranchDataProvider, WorkspaceResource as v2WorkspaceResource } from '../v2AzureResourcesApi';
+
+export class CompatibleWorkspaceResourceBranchDataProvider<TResource extends WorkspaceResource & v2WorkspaceResource> extends AzExtTreeDataProvider implements BranchDataProvider<TResource, TResource> {

```

Comment:
This is largely identical to `CompatibleBranchDataProvider`, seems like we should refactor to share code. I'm not sure if we need a workspace- or application- specific one at all.